edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java

```diff
@@ -3785,4 +3915,55 @@ private void excludeUncategorisedItemsFromCourseGradeCalculations(final Gradeboo
         
 	}
 
+	/**
+	 *  Toggles the excluded status for an assignment grade record
+	 */
+	public void toggleGradeRecordExcluded(final String gradebookUid, final Long assignmentId, final String studentUid){
+		GradebookAssignment assignment = getAssignmentWithoutStats(gradebookUid, assignmentId);
+		AssignmentGradeRecord gradeRecord = getAssignmentGradeRecord(assignment, studentUid);
+		if(gradeRecord != null) {
+			if (gradeRecord.isExcludedFromGrade() == null) {
+				gradeRecord.setExcludedFromGrade(true);
+			} else if (gradeRecord.isExcludedFromGrade()) {
+				gradeRecord.setExcludedFromGrade(false);
+			} else {
+				gradeRecord.setExcludedFromGrade(true);
+			}
+		}
+	}
+
+	/**
+	 * Checks if an assignment is excluded in the AGR for an individual student
+	 */
+	public boolean getExculdedStatusForAssignmentByStudent(final String gradebookUid, final Long assignmentId, final String studentUid){
```

Comment:
This is probably unnecessary. If you can get the gradeRecord you can just check the value. It doesn't need to be in a public API. You could create a helper in the AssignmentGradeRecord class or update the getter